go/vt/binlog/keyrange_filter.go

```diff
@@ -40,40 +30,23 @@ func KeyRangeFilterFunc(kit key.KeyspaceIdType, keyrange *pb.KeyRange, sendReply
 				log.Warningf("Not forwarding DDL: %s", statement.Sql)
 				continue
 			case proto.BL_DML:
-				keyspaceIndex := strings.LastIndex(statement.Sql, KEYSPACE_ID_COMMENT)
-				if keyspaceIndex == -1 {
+				keyspaceId, unfriendly, err := annotations.ParseSQLAnnotation(string(statement.Sql))
+				if err != nil {
 					updateStreamErrors.Add("KeyRangeStream", 1)
-					log.Errorf("Error parsing keyspace id: %s", statement.Sql)
+					log.Errorf(
+						"Error parsing keyspace id annotation. Skipping statement: %s, (%s)",
+						string(statement.Sql), err)
 					continue
 				}
-				idstart := keyspaceIndex + len(KEYSPACE_ID_COMMENT)
-				idend := strings.Index(statement.Sql[idstart:], SPACE)
-				if idend == -1 {
+				if unfriendly {
 					updateStreamErrors.Add("KeyRangeStream", 1)
-					log.Errorf("Error parsing keyspace id: %s", statement.Sql)
+					log.Errorf(
```

Comment:
If we skip DMLs, it's likely that the diff will later fail anyway. Maybe it would be better to halt filtered replication instead of skipping? Alternatively, @alainjobart mentioned that some statements might work ok if they're broadcast to all destination shards. In that case at least there's a chance the diff might come up clean, as opposed to skipping the DML altogether.